### ISSUE TITLE
Error "Scan"

### ISSUE THREAD

**Issue by danieljohnmorris** [DamirSvrtan/fasterer#8]
```
$ fasterer
app/controllers/data/charts_controller.rb
For loop is slower than using each. Occured at lines: 76, 87, 90, 102, 206, 240.

app/controllers/data_controller.rb
For loop is slower than using each. Occured at lines: 119, 146, 147.

app/models/country.rb
For loop is slower than using each. Occured at lines: 22, 26.

app/models/event.rb
For loop is slower than using each. Occured at lines: 445, 459, 472, 486, 498, 614.

app/models/user.rb
Calling argumentless methods within blocks is slower than using symbol to proc. Occured at lines: 21, 31.

/Users/dan/.rvm/gems/ruby-2.2.0/gems/ruby_parser-3.6.4/lib/ruby_lexer.rex.rb:288:in `scan': execution expired (Timeout::Error)
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/ruby_parser-3.6.4/lib/ruby_lexer.rex.rb:288:in `next_token'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/ruby_parser-3.6.4/lib/ruby_parser_extras.rb:919:in `next_token'
	from /Users/dan/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/racc/parser.rb:258:in `_racc_do_parse_c'
	from /Users/dan/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/racc/parser.rb:258:in `do_parse'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/ruby_parser-3.6.4/lib/ruby_parser_extras.rb:1036:in `block in process'
	from /Users/dan/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:89:in `block in timeout'
	from /Users/dan/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:34:in `block in catch'
	from /Users/dan/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:34:in `catch'
	from /Users/dan/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:34:in `catch'
	from /Users/dan/.rvm/rubies/ruby-2.2.0/lib/ruby/2.2.0/timeout.rb:104:in `timeout'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/ruby_parser-3.6.4/lib/ruby_parser_extras.rb:1024:in `process'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/ruby_parser-3.6.4/lib/ruby_parser_extras.rb:1357:in `block in process'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/ruby_parser-3.6.4/lib/ruby_parser_extras.rb:1355:in `each'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/ruby_parser-3.6.4/lib/ruby_parser_extras.rb:1355:in `process'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/fasterer-0.1.7/lib/fasterer/parser.rb:8:in `parse'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/fasterer-0.1.7/lib/fasterer/analyzer.rb:21:in `scan'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/fasterer-0.1.7/lib/fasterer/file_traverser.rb:56:in `scan_file'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/fasterer-0.1.7/lib/fasterer/file_traverser.rb:67:in `block in traverse_directory'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/fasterer-0.1.7/lib/fasterer/file_traverser.rb:64:in `each'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/fasterer-0.1.7/lib/fasterer/file_traverser.rb:64:in `traverse_directory'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/fasterer-0.1.7/lib/fasterer/file_traverser.rb:27:in `traverse'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/fasterer-0.1.7/lib/fasterer/cli.rb:7:in `execute'
	from /Users/dan/.rvm/gems/ruby-2.2.0/gems/fasterer-0.1.7/bin/fasterer:5:in `<top (required)>'
	from /Users/dan/.rvm/gems/ruby-2.2.0/bin/fasterer:23:in `load'
	from /Users/dan/.rvm/gems/ruby-2.2.0/bin/fasterer:23:in `<main>'
	from /Users/dan/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `eval'
	from /Users/dan/.rvm/gems/ruby-2.2.0/bin/ruby_executable_hooks:15:in `<main>'
```

**Comment by DamirSvrtan** [comment]
Hey @danieljohnmorris, I've released a new version (0.1.8) on Rubygems that rescues the `Timeout::Error`. Since this is raised by the internal parser, I can't do much about it except rescuing it.
Please try again and tell me if some problems persist.

**Comment by DamirSvrtan** [comment]
@danieljohnmorris I'm closing this, if any other errors come up please open an issue.
